pingrid.py

```diff
@@ -770,3 +771,23 @@ def _proxy(fn, *args, **kwargs):
     if decode_times:
         ds = fix_calendar(ds)
     return ds
+
+
+def targetRangeFormat(leadsValues,leadsKeys,startDate,periodLength):
```

Comment:
I did not know if it was better to have the inputs be the separate lists of lead values / keys like this, or if it is better to have the input be the dictionary and then within the function, have the code that creates the lists. Or if it makes much of a difference either way.

Reply:
Personally I would choose neither of those options. I would have a function that formats a single date range, and call that function once for each value in the dictionary.

Reply:
Additionally, there is also a getTargets function that is very similar in predictions.py. I think there is some rationalization to do with those 2 functions.   Maybe something like one function that gets target_start and target_end from issue, lead and length. Another that formats dates, whether they are a single date or a period described by a start and end.

Reply:
Latest commits address these comments. I thought about making the `target_range_format` function return the target start and target end separately instead of as a single string, what do you think? Or leave it as returning as a single string?

Reply:
I think we almost never will need target_start and target_end separately so I am not sure there is added value outputing them separately. I think we will need at some point to split what calculates the datetime objects and what format them for printing. But we can deal with that later when concrete cases arise.